Improve webhook security with prepared statements and input validation

ndabene opened the pull request.

ndabene committed:
Improve webhook security with prepared statements and input validation

prestaalba commented on classes/Webhook.php:
```diff
@@ -38,12 +38,28 @@ class Webhook extends ObjectModel
+        }
+
+        // Security: Use prepared query to prevent SQL injection
         $sql = 'SELECT id_webhook
```
Original query already prevent SQL injection.

ndabene commented on ps_webhooks.php:
```diff
@@ -117,10 +140,12 @@ public static function executeUrl($url, $action, $entity, $object, $test = false
             CURLOPT_TIMEOUT => 3,
-            CURLOPT_SSL_VERIFYHOST => false,
-            CURLOPT_SSL_VERIFYPEER => false,
+            // Security: Enable SSL verification to prevent MITM attacks
```
You are true, I always use SSL on my dev shop :)

ndabene committed:
Upgrade SQL query to DbQuery object for improved security and maintainability

ndabene committed:
Remove redundant security checks in Webhook::getIdsByActionEntity

The security validations for hook parameter and SQL injection prevention were removed as they are no longer needed, simplifying the method while maintaining functionality.

ndabene committed:
Enable conditional SSL verification based on HTTPS configuration to improve security and compatibility with proxy setups

prestaalba commented:
Thank your for your contribution!!!
